src/ctapipe/io/simteleventsource.py

```diff
@@ -1085,11 +1113,79 @@ def _fill_mon_pixels_status(self, tel_id):
         disabled_pixels = np.zeros((n_gains, n_pixels), dtype=bool)
         disabled_pixels[:, disabled_ids] = True
 
-        return PixelStatusContainer(
-            hardware_failing_pixels=disabled_pixels,
-            pedestal_failing_pixels=disabled_pixels.copy(),
-            flatfield_failing_pixels=disabled_pixels.copy(),
+        return disabled_pixels
+
+    def _get_monitoring_data(self):
```

Comment:
Implementing the IO for monitoring data directly in `SimTelEventSource` is not a good fit.

This should be independent from `SimTelEventSource` (see comment above). It should work the same for other event sources (at least the HDF5EventSource also needs support).

But as said, I think this is best implemented as a step after event source:

so that we have in `ctapipe-process` something like this:
```
event_source = EventSource(...)
calibration_source = CalibrationSource.from_name(self.calibration_source_type, ...)

for event in event_source:
    calibration_source(event)
```

Reply:
Done